code/game/objects/structures/barsign.dm

```diff
@@ -47,68 +52,141 @@
-		else
-			to_chat(user, "<span class='warning'>You need at least two lengths of cable!</span>")
+	if(isscrewdriver(I) || iswrench(I))
+		return	// No damage.
 	else
 		return ..()
 
+/obj/structure/sign/barsign/screwdriver_act(mob/user)
+	if(!panel_open)
+		to_chat(user, "<span class='notice'>You open the maintenance panel.</span>")
+		set_sign(new /datum/barsign/signoff)
+		panel_open = TRUE
+	else
+		if(prev_sign.icon != icon_state && istype(prev_sign, /datum/barsign))	// Sanity, I suppose
+			set_sign(prev_sign)
+		to_chat(user, "<span class='notice'>You close the maintenance panel.</span>")
+		panel_open = FALSE
+	return
+
+/obj/structure/sign/barsign/wrench_act(mob/user)
+	if(!panel_open)
+		to_chat(user, "<span_class='notice>You must first open the maintenance panel before trying to unbolt the barsign.</span>")
+		return
+	else
+		var/obj/item/sign/barsign/S = new(user.loc)
+		S.name = name
+		S.desc = desc
+		S.icon_state = icon_state	//The only sprite direction that exists is South.
+		S.broken = broken
+		S.emagged = emagged
+		S.req_access = req_access
+		S.panel_open = !panel_open
+		S.prev_sign = prev_sign
+		qdel(src)
+
+/obj/item/sign/barsign
+	name = "barsign"
+	desc = ""
+	icon = 'icons/obj/barsigns.dmi'
+	w_class = WEIGHT_CLASS_HUGE		// Don't put this in your backpack!
+	resistance_flags = FLAMMABLE
+	var/panel_open
+	var/broken
+	var/datum/barsign/prev_sign
+
+/obj/item/sign/barsign/wrench_act(mob/user)	//construction
+	if(isturf(user.loc))
+		var/direction = input("In which direction?", "Select direction.") in list("North", "East", "South", "West", "Cancel")
+		//The only sprite that exists is South, but we forced this in when we dismantled the barsign.
+		if(direction == "Cancel")
+			return
+		if(QDELETED(src))
+			return
+		var/obj/structure/sign/barsign/S = new(user.loc)
+		switch(direction)
+			if("North")
+				S.pixel_y = 32
+			if("East")
+				S.pixel_x = 32
+			if("South")
+				S.pixel_y = -32
+			if("West")
+				S.pixel_x = -32
+			else
+				return
+		S.name = name
+		S.desc = desc
+		S.broken = broken
+		S.emagged = emagged
+		S.req_access = req_access
+		S.panel_open = panel_open
+		S.prev_sign = prev_sign
+		S.icon_state = prev_sign.icon
+		S.sign_holder = null
+		to_chat(user, "<span_class='notice>You bolt \the [S] with your wrench, closing the maintenance panel in the process.</span>")
+		qdel(src)
+	else
+		return
 
+/obj/structure/sign/barsign/proc/pick_sign()
+	var/list/signs = barsigns.Copy()
+	var/new_sign
+	if(!broken)
+		if(!emagged)
+			new_sign = input("Available Signage: ", "Bar Sign", null) as null|anything in signs
+			set_sign(new_sign)
+		else
+			signs += hiddensigns
+			new_sign = input("Available Signage: ", "Bar Sign", null) as null|anything in signs
+			set_sign(new_sign)
+	else
+		set_sign(new /datum/barsign/empbarsign)
 
 /obj/structure/sign/barsign/emp_act(severity)
-    set_sign(new /datum/barsign/hiddensigns/empbarsign)
-    broken = 1
-
-
-
+	set_sign(new /datum/barsign/empbarsign)
```

Comment:
Probably no need to set this over and over if it's already `broken`, in case of multiple EMPs.